monai/data/meta_tensor.py

```diff
@@ -565,7 +572,7 @@ def ensure_torch_and_prune_meta(
         if simple_keys:
             # ensure affine is of type `torch.Tensor`
             if MetaKeys.AFFINE in meta:
-                meta[MetaKeys.AFFINE] = convert_to_tensor(meta[MetaKeys.AFFINE])  # bc-breaking
+                meta[MetaKeys.AFFINE] = convert_to_tensor(meta[MetaKeys.AFFINE], device=device)  # bc-breaking
```

Comment:
May don't need move affine to device here.

Reply:
Hi @KumoLiu , thanks for the comments, I removed the device arg, and re-tested the changes with decathlon Task03_Liver dataset.